### DISCUSSION TITLE
Memory Storage on Mobile Devices

### DISCUSSION THREAD

**Discussion by maioradv** [arthurfiorette/axios-cache-interceptor#621] [Q&A] [answered]
Hi everyone, i'm using axios-cache-interceptor to cache axios request on a React Native Expo app, i was wondering where the data are stored on the device (RAM?) and if there is a limit... anyway an explanation how this memory storage works on android and ios.

**Comment by arthurfiorette** [comment, marked as answer]
Hey @maioradv, memory storage just saves everything on a js variable, without any persistent storage or else. I guess its up to how many RAM the user's device have. And it will reset every time the user open/close the app.

If you're running expo, [`Device.getMaxMemoryAsync()`](https://docs.expo.dev/versions/latest/sdk/device/#devicegetmaxmemoryasync)_(androind only)_ or [`Device.totalMemory`](https://docs.expo.dev/versions/latest/sdk/device/#devicegetmaxmemoryasync) will get you the exact values of how many ram is available at runtime.

However, if you're thinking about RAM usage/limit, probably you should use a persistent storage and save cache on the filesystem, but its up to your use case.
